src/FieldWarning/Assets/Units/Component/Vision/VisionComponent.cs

```diff
@@ -111,8 +114,38 @@ private bool CanDetect(VisionComponent target)
             return
                 distance < max_spot_range
                 && distance < max_spot_range * stealth_pen_factor / target.stealth_factor;
+                
         }
+        private bool ObstacleInLineOfSight(VisionComponent spotter, VisionComponent target)
+        {
 
+            int layerMask = 1 << 13;
+            Vector3 SpotterPosition = spotter.gameObject.transform.position;
+            Vector3 temp = SpotterPosition;
```

Comment:
A comment about why this workaround was necessary will help us a lot when we're reading this code 3 months from now. Also you don't need the intermediate variable, just do 

SpotterPosition = new Vector3(SpotterPosition.x, SpotterPosition.y + 0.5, SpotterPosition.z);